src/adapters/dummy/mod.rs

```diff
@@ -84,6 +85,24 @@ pub struct DummyAdapter<T> {
     controller: T
 }
 
+struct DummyListener;
+
+impl DummyListener {
+    pub fn new() -> Arc<Mutex<Self>> {
+        Arc::new(Mutex::new(DummyListener))
```

Comment:
Maybe it doesn't need to be put in a mutex, similar to how controller and upnp work? (Any data would be protected inside the callback methods of the trait by their own mutex?)